src/Autofac/Util/ReflectionExtensions.cs

```diff
@@ -121,4 +121,54 @@ public static bool TryGetDeclaringProperty(this ParameterInfo pi, [NotNullWhen(r
 
         return callExpression.Constructor!;
     }
+
+    /// <summary>
+    /// Checks if a provided member has a <c>RequiredMemberAttribute</c>.
+    /// </summary>
+    /// <remarks>
+    /// <para>
+    /// On NET7+ this would <em>typically</em> be the framework supplied <c>RequiredMemberAttribute</c>, <em>but</em> internally the compiler
+    /// <em>only</em> requires an attribute with that specific type <em>name</em>, not that specific type <em>reference</em>.
+    /// </para>
+    /// <para>
+    /// This could very well be an internally defined custom polyfill attribute using that type name, so this
+    /// check is done <em>only</em> via type <em>name</em>, not reference.
+    /// </para>
+    /// </remarks>
+    /// <param name="memberInfo">Member to check.</param>
+    /// <returns>
+    /// <see langword="true" /> if <paramref name="memberInfo"/> carries a <see cref="MemberInfo.CustomAttributes">CustomAttributeData</see> with
+    /// a type <em>name</em> of <c>System.Runtime.CompilerServices.RequiredAttribute</c>; <see langword="false" /> otherwise.
+    /// </returns>
+    public static bool HasRequiredMemberAttribute(
+        this MemberInfo memberInfo)
+    {
+        return memberInfo.CustomAttributes.Any(
+            cad => cad.AttributeType.FullName == "System.Runtime.CompilerServices.RequiredMemberAttribute");
+    }
+
+    /// <summary>
+    /// Checks if a constructor has a <c>SetsRequiredMembersAttribute</c>.
+    /// </summary>
+    /// <remarks>
+    /// <para>
+    /// On NET7+ this would <em>typically</em> be the framework supplied <c>SetsRequiredMembersAttribute</c>, <em>but</em> internally the compiler
+    /// <em>only</em> requires an attribute with that specific type <em>name</em>, not that specific type <em>reference</em>.
+    /// </para>
+    /// <para>
+    /// This could very well be an internally defined custom polyfill attribute using that type name, so this
+    /// check is done <em>only</em> via type <em>name</em>, not reference.
+    /// </para>
+    /// </remarks>
+    /// <param name="constructorInfo">Constructor to check.</param>
+    /// <returns>
+    /// <see langword="true" /> if <paramref name="constructorInfo"/> carries a <see cref="MemberInfo.CustomAttributes">CustomAttributeData</see> with
+    /// a type <em>name</em> of <c>System.Diagnostics.CodeAnalysis.SetsRequiredMembersAttribute</c>; <see langword="false" /> otherwise.
+    /// </returns>
+    public static bool HasSetsRequiredMembersAttribute(
+        this ConstructorInfo constructorInfo)
```

Comment:
Doesn't `ConstructorInfo` derive from `MemberInfo`? Do you need the overload?

Reply:
Checks for a different attribute: `SetsRequiredMembersAttribute`, not `RequiredMemberAttribute` (and only ever does so for constructors).

Reply:
Of course, my bad 🤦